### PR TITLE
846 disable rerun for instruments with novariables

### LINKED ISSUE #846: Rerun not possible for runs that do not have variables.
Issue raised by: [developer/user/project requirement]

### What?
It is not possible to rerun a job for an instrument that does not have variables.
We should either hide the buttons for rerunning for an instrument like this. Or allow a rerun to be submitted for this scenario.

Currently this only effects WISH who do not use the WebApp anyway.

### How?
Smoke testing
### Reproducible?
[Yes]

### How to test the issue is resolved
Either button is not displayed for these instruments or it is possible to rerun these jobs.

### PR DESCRIPTION
**Pull request by keiranjprice101**
### Summary of work
Removes the option to rerun jobs through the WebApp if an instrument does not have variables. (As is currently the case only for WISH)

### How to test your work
 
Check that the rerun option in the run-summary and instrument-summary do not appear for WISH only.

### Additional comments
 

closes #846

**Before merging ensure the release notes have been updated**

### CODE REVIEW

File: WebApp/autoreduce_webapp/reduction_viewer/views.py [hunk 1]
```diff
@@ -339,6 +342,8 @@ def instrument_summary(request, instrument=None):
         if len(runs) == 0:
             return {'message': "No runs found for instrument."}
 
+        has_variables = InstrumentVariablesUtils().get_default_variables(instrument_obj.name)
```
Comment (Anders-Markvardsen) [inline WebApp/autoreduce_webapp/reduction_viewer/views.py:345]: Is there a bool() missing here around InstrumentVariablesUtils().get_default_variables(instrument_obj.name)?

File: WebApp/autoreduce_webapp/reduction_viewer/views.py [hunk 2]
```diff
@@ -273,12 +272,16 @@ def run_summary(_, instrument_name=None, run_number=None, run_version=0):
             reduction_location = reduction_location.replace('\\', '/')
 
         rb_number = Experiment.objects.get(id=run.experiment_id).reference_number
+        has_variables = bool(
```
Comment (Anders-Markvardsen) [inline WebApp/autoreduce_webapp/reduction_viewer/views.py:275]: Could you add a comment explaining why you check for both InstrumentVariablesUtils().get_default_variables(run.instrument.name) and run.run_variables.all() ?

File: WebApp/autoreduce_webapp/reduction_viewer/views.py [hunk 3]
```diff
@@ -16,25 +16,24 @@
...
-                                          require_admin, check_permissions)
-from plotting.plot_handler import PlotHandler
-from reduction_variables.utils import MessagingUtils
+from reduction_variables.utils import InstrumentVariablesUtils, MessagingUtils
 from reduction_viewer.models import Experiment, ReductionRun, Instrument, Status
 from reduction_viewer.utils import StatusUtils, ReductionRunUtils
 from reduction_viewer.view_utils import deactivate_invalid_instruments
 from utilities.pagination import CustomPaginator
 
+from plotting.plot_handler import PlotHandler
```
Comment (Anders-Markvardsen) [inline WebApp/autoreduce_webapp/reduction_viewer/views.py:36]: Is the reason for moving the order of the above headers round to have them listed in alphabetic order?

**Review by Anders-Markvardsen — CHANGES_REQUESTED** [review]
Just received the code in this review. See comments

File: WebApp/autoreduce_webapp/reduction_viewer/views.py (hunk 3 above)
Comment (keiranjprice101) [reply WebApp/autoreduce_webapp/reduction_viewer/views.py:36]: This was automatic by IDE.. I think the reason the order seems strange is because of #763

**Review by Anders-Markvardsen — APPROVED** [review]

**Pull request merged**